### PR TITLE
[ONNX] embedding

### PR DESCRIPTION
**Pull request by cynthia-liu**

### CODE REVIEW

**Comment by cynthia-liu** [comment]
http://10.239.47.211:18888/view/ZOO-PR/job/ZOO-PR-Validation/1400/

File: pyzoo/test/zoo/pipeline/utils/test_utils_onnx.py [hunk 1]
```diff
@@ -52,14 +55,18 @@ def _convert_ndarray_to_tensor(self, input_data_with_batch):
         else:
             return tensors
 
-    def compare_with_pytorch(self, pytorch_model, input_shape_with_batch,
+    def compare_with_pytorch(self, pytorch_model, input_shape_with_batch, input_data_with_batch=0,
```
Comment (zhichao-li) [inline pyzoo/test/zoo/pipeline/utils/test_utils_onnx.py]: change the default value to `None`

File: pyzoo/test/zoo/pipeline/utils/test_utils_onnx.py [hunk 2]
```diff
@@ -52,14 +55,18 @@ def _convert_ndarray_to_tensor(self, input_data_with_batch):
...
                              compare_result=True, rtol=1e-6, atol=1e-6):
         input_shape_with_batch = to_list(input_shape_with_batch)
-        onnx_path = self.dump_pytorch_to_onnx(pytorch_model, input_shape_with_batch)
+        onnx_path = self.dump_pytorch_to_onnx(pytorch_model, input_shape_with_batch,
+                                              input_data_with_batch)
```
Comment (zhichao-li) [inline pyzoo/test/zoo/pipeline/utils/test_utils_onnx.py]: `input_data_with_batch` should be treated as a list which is similar with input_shape_with_batch

File: pyzoo/zoo/pipeline/api/onnx/mapper/gather.py [hunk 3]
```diff
@@ -28,12 +28,16 @@ def _extract_model_inputs(self):
...
-
-        dim = int(self.onnx_attr['axis'])
-        assert dim >= 1, "Currently only dim>=1 is supported."
-
         indices = self.model_inputs[1].zvalue
-        assert indices.shape == (1,), "Currently only one index is supported."
-        index = int(indices.get_weight().max())
 
-        return zautograd.expand_dims(data.index_select(dim=dim, index=index), axis=dim)
+        if "axis" in self.onnx_attr.keys():
```
Comment (zhichao-li) [inline pyzoo/zoo/pipeline/api/onnx/mapper/gather.py]: you should use the type of `data` for conditional switching instead of `axis` existing or not.

**Comment by cynthia-liu** [comment]
http://10.239.47.211:18888/view/ZOO-PR/job/ZOO-PR-Validation/1408/

**Pull request merged**